src/coreclr/src/vm/peimage.cpp

```diff
@@ -973,10 +973,7 @@ PTR_PEImageLayout PEImage::GetLayoutInternal(DWORD imageLayoutMask,DWORD flags)
         BOOL bIsFlatLayoutSuitable = ((imageLayoutMask & PEImageLayout::LAYOUT_FLAT) != 0);
 
 #if !defined(TARGET_UNIX)
-        if (bIsMappedLayoutSuitable)
-        {
-            bIsFlatLayoutSuitable = FALSE;
-        }
+        bIsFlatLayoutSuitable = IsInBundle() || !bIsMappedLayoutSuitable;
```

Comment:
This doesn't look right. We are completely discarding the previously computed value of the `bIsFlatLayoutSuitable`.